File: 2025_08/leetcode2025-08-09.cpp
#include <iostream>
using namespace std;

class Solution {
public:
    bool isPowerOfTwo(int n) {
        return n==0 ? false : (long long)n==((long long)n&(-(long long)n));
    }
};

int main(){
    Solution s;
    int n = 1;
    cout << s.isPowerOfTwo(n) << endl;
}

File: 2025_08/leetcode2025-08-09.py
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        # 判断该数是否等于它最后一位1即可
        return False if n==0 else n==(n&(-n))

if __name__ == "__main__":
    s = Solution()
    n = 1
    print(s.isPowerOfTwo(n))
